Code/src/main/java/nl/utwente/group10/ui/InspectorWindow.java

```diff
@@ -75,11 +75,11 @@ public void update() {
     /**
      * Walks the expr tree, walk recursively calls itself on its children.
      */
-    private void walk(TreeItem<String> treeItem, Expr expr) {
+    private void walk(TreeItem<String> treeItem, Expression expr) {
         String type;
 
         try {
-            type = expr.getType(pane.getEnvInstance()).toHaskellType();
+            type = expr.findType().toHaskellType();
```

Comment:
Does `findType` change the expression in any way/do any inference? If so, this change might cause additional confusion (in the category 'it only works if I have the Inspector open').